packages/blade/src/components/Dropdown/Dropdown.tsx

```diff
@@ -74,11 +76,26 @@ const _Dropdown = ({
   const dropdownBaseId = useId('dropdown');
 
   const dropdownTriggerer = React.useRef<DropdownContextType['dropdownTriggerer']>();
+  const isFirstRenderRef = React.useRef(true);
+
+  React.useEffect(() => {
+    // Ignoring the `onDismiss` call on first render
+    if (isFirstRenderRef.current) {
+      isFirstRenderRef.current = false;
+      return;
+    }
+
+    if (!isOpen && onDismiss) {
+      onDismiss();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen]);
 
   React.Children.map(children, (child) => {
     if (React.isValidElement(child)) {
       if (
         !isValidAllowedChildren(child, 'SelectInput') &&
+        !isValidAllowedChildren(child, 'DropdownButton') &&
```

Comment:
let's refactor out all these component ids into the componentIds constant.

Reply:
done